File: 백준/Silver/2960. 에라토스테네스의 체/에라토스테네스의 체.py
import sys

N, K = map(int, sys.stdin.readline().split(' '))
num_list = []
remove = []

for i in range(2, N+1):
    num_list.append(i)
    remove.append(0)

cnt = 0
idx = 0
while True:
    for i in range(len(num_list)):
        P = num_list[idx]

        if (num_list[i] % P == 0):
            if (remove[i] == 0):
                cnt += 1
                remove[i] = 1

            if (cnt == K):
                print(num_list[i])
                exit(0)
    idx += 1

File: 백준/Silver/2960. 에라토스테네스의 체/에라토스테네스의 체.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        int N = Integer.parseInt(st.nextToken());
        int K = Integer.parseInt(st.nextToken());

        int[] arr = new int[N - 1];
        int[] remove = new int[N - 1];

        for (int i = 0; i < N - 1; i++) {
            arr[i] = i + 2;
            remove[i] = 0;
        }

        int cnt = 0;
        int idx = 0;

        while (cnt != K) {
            for (int i = 0; i < arr.length; i++) {
                if (remove[i] == 0) {
                    int P = arr[idx];

                    if (arr[i] % P == 0) {
                        cnt += 1;
                        remove[i] = 1;
                    }

                    if (cnt == K) {
                        System.out.println(arr[i]);
                        break;
                    }
                }
            }
            idx += 1;
        }

    }
}
